src/S3/Transfer.php

```diff
@@ -305,6 +324,8 @@ private function upload($filename)
         $args = $this->s3Args;
         $args['SourceFile'] = $filename;
         $args['Key'] = $this->createS3Key($filename);
+        $this->inProgress['FileSource'] = $args['Bucket'] . "/" . $args['Key'];
```

Comment:
There are actually N number of transfers in progress at any given time. When a failure occurs, the transfer stops at the point of failure, leaving the transfer in a state in which there are N number of transfers that are incomplete. I believe that when you wait on the Transfer it will throw when the first error is encountered, leaving the other currently in-progress transfers incomplete. You might want to double check the behavior here. If that's the case, then you would need to provide context on all of the in-progress transfers *and* the failed transfer when throwing the exception.